pkg/virt-handler/vm.go

```diff
@@ -181,13 +196,18 @@ func (d *VMHandlerDispatch) Execute(store cache.Store, queue workqueue.RateLimit
 	logging.DefaultLogger().V(3).Info().Object(vm).Msg("Processing VM update.")
 
 	// Process the VM
-	err = d.processVmUpdate(vm, shouldDeleteVm)
+	isPending, err := d.processVmUpdate(vm, shouldDeleteVm)
 	if err != nil {
 		// Something went wrong, reenqueue the item with a delay
 		logging.DefaultLogger().Error().Object(vm).Reason(err).Msg("Synchronizing the VM failed.")
 		d.recorder.Event(vm, k8sv1.EventTypeWarning, v1.SyncFailed.String(), err.Error())
 		queue.AddRateLimited(key)
 		return
+	} else if isPending {
+		// waiting on an async action to complete
+		logging.DefaultLogger().V(3).Info().Object(vm).Reason(err).Msg("Synchronizing is in a pending state.")
+		queue.AddRateLimited(key)
```

Comment:
I think you should use something like `queue.AddAfter(key, 1 * time.Second)`, followed by `queue.Forget(key)`, to cleare the rate limiter.

If we reach this point we had a successful controller run, we are just not yet done.

Reply:
yup, good idea. i fixed this.